tests/PuliRunnerTest.php

```diff
@@ -13,65 +13,88 @@
 
 use PHPUnit_Framework_TestCase;
 use Puli\ComposerPlugin\PuliRunner;
-use ReflectionObject;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\ProcessUtils;
 use Webmozart\PathUtil\Path;
 
 /**
  * @since  1.0
  *
  * @author Titouan Galopin <galopintitouan@gmail.com>
+ * @author Bernhard Schussek <bschussek@gmail.com>
  */
 class PuliRunnerTest extends PHPUnit_Framework_TestCase
 {
+    /**
+     * @var string
+     */
+    private $fixturesDir;
+
+    /**
+     * @var string
+     */
+    private $previousWd;
+
+    protected function setUp()
+    {
+        $this->fixturesDir = Path::normalize(__DIR__.'/Fixtures/scripts');
+        $this->previousWd = getcwd();
+    }
+
+    protected function tearDown()
+    {
+        chdir($this->previousWd);
+    }
+
     public function testRunnerBash()
     {
-        $fixturesDir = Path::normalize(__DIR__.'/Fixtures/scripts/bash');
-        $runner = new PuliRunner($fixturesDir);
+        chdir($this->fixturesDir.'/bash');
+
+        $runner = new PuliRunner();
 
-        $this->assertRunnerUseScript($fixturesDir, $runner, false);
+        $expected = escapeshellcmd($this->fixturesDir.'/bash/puli');
+
+        $this->assertSame(strtr($expected, '/', DIRECTORY_SEPARATOR), $runner->getPuliCommand());
     }
 
-    public function testRunnerPhpClassical()
+    public function testRunnerBat()
     {
-        $fixturesDir = Path::normalize(__DIR__.'/Fixtures/scripts/php_classical');
-        $runner = new PuliRunner($fixturesDir);
+        if ('\\' !== DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('Requires Windows to run');
+        }
 
-        $this->assertRunnerUseScript($fixturesDir, $runner, true);
+        chdir($this->fixturesDir.'/bat');
+
+        $runner = new PuliRunner();
+
+        $expected = escapeshellcmd($this->fixturesDir.'/bat/puli.bat');
+
+        $this->assertSame(strtr($expected, '/', DIRECTORY_SEPARATOR), $runner->getPuliCommand());
     }
 
-    public function testRunnerPhpHashbang()
+    public function testRunnerPhpClassical()
     {
-        $fixturesDir = Path::normalize(__DIR__.'/Fixtures/scripts/php_hashbang');
-        $runner = new PuliRunner($fixturesDir);
+        chdir($this->fixturesDir.'/php_classical');
+
+        $runner = new PuliRunner();
+        $phpFinder = new PhpExecutableFinder();
+        $php = $phpFinder->find();
 
-        $this->assertRunnerUseScript($fixturesDir, $runner, true);
+        $expected = escapeshellcmd($php).' '.ProcessUtils::escapeArgument($this->fixturesDir.'/php_classical/puli');
```

Comment:
``escapeshellcmd`` does not escape spaces inside the executable path, as it will accept them as argument separator (as you escape the full command rather than each argument). This may not be what you expect.